docs/user/advanced.rst

```diff
@@ -64,26 +64,21 @@ Some applications render templates based on data sources (e.g. CSVs or
 JSON files).
 
 To get data to templates you can set up a mapping between filenames
-and functions which generate dictionaries containing the data:
+and dictionaries or functions which generate dictionaries containing 
+the data:
```

Comment:
I'm going to change the language here to read "set up a mapping between filenames and either dictionaries or functions that generate dictionaries containing the data" since I think that's a little more clear.